project/SbtScriptedIT.scala

```diff
@@ -0,0 +1,106 @@
+import sbt._
+import Keys._
+
+import java.io.File
+import java.util.UUID.randomUUID
+
+object SbtScriptedIT extends AutoPlugin {
+
+  object autoImport {
+    val scriptedTestSbtRepo = settingKey[String]("SBT repository to be used in scripted tests")
+    val scriptedTestSbtRef = settingKey[String]("SBT branch to be used in scripted tests")
+    val scriptedTestLMImpl = settingKey[String]("Librarymanagement implementation to be used in scripted tests")
+  }
+
+  import autoImport._
+  override def requires = ScriptedPlugin
+
+  override def trigger = noTrigger
+
+  override lazy val globalSettings = Seq(
+    scriptedTestSbtRepo := "https://github.com/sbt/sbt.git",
+    scriptedTestSbtRef := "develop",
+    scriptedTestLMImpl := "ivy"
+  )
+
+  private def cloneSbt(targetDir: File, repo: String, ref: String) = {
+    import org.eclipse.jgit.api._
+
+    if (!targetDir.exists) {
+      IO.createDirectory(targetDir)
+
+      new CloneCommand()
+        .setDirectory(targetDir)
+        .setURI(repo)
+        .call()
+
+      val git = Git.open(targetDir)
+
+      git.checkout().setName(ref).call()
+    }
+  }
+
+  private def publishLocalSbt(
+      targetDir: File,
+      lmVersion: String,
+      lmGroupID: String,
+      lmArtifactID: String,
+      version: String) = {
+    import sys.process._
+    Process(
+      Seq(
+        "sbt",
+        "-J-Xms2048m",
+        "-J-Xmx2048m",
+        "-J-XX:ReservedCodeCacheSize=256m",
+        "-J-XX:MaxMetaspaceSize=512m",
+        s"""-Dsbt.librarymanagement.version=${lmVersion}""",
+        s"""-Dsbt.librarymanagement.groupID=${lmGroupID}""",
+        s"""-Dsbt.librarymanagement.artifactID=${lmArtifactID}""",
+        s"""set ThisBuild / version := "${version}"""",
+        "clean",
+        "publishLocal"
+      ),
+      Some(targetDir)
+    ) !
+  }
+
+  private def setScriptedTestsSbtVersion(baseDir: File, version: String) = {
+    IO.listFiles(baseDir).foreach { d =>
+      if (d.isDirectory) {
+        IO.createDirectory(d / "project")
+        IO.write(
+          d / "project" / "build.properties",
+          s"sbt.version=$version"
+        )
+      }
+    }
+  }
+
+  private def sbtUniqueVersion(sbtBaseVersion: String) =
+    s"""$sbtBaseVersion-${randomUUID.toString.replace("-","")}-SNAPSHOT"""
```

Comment:
This might end up accumulating too much JAR on cache directories. I think we should hard code to `LM-SNAPSHOT` and also clean it up afterwards on Travis CI.

Reply:
fixed with a `settingKey` and a clean job 'before-cache' in travis, I left commented out an extra clean of `local` since it's not cached after all.